vortex-sampling-compressor/src/compressors/fsst.rs

```diff
@@ -91,18 +92,45 @@ impl EncodingCompressor for FSSTCompressor {
                 like.as_ref().and_then(|l| l.child(0)),
             )?;
 
+        let codes_compressor = ctx.auxiliary("codes");
+        let codes_varbin = VarBinArray::try_from(fsst_array.codes())?;
+        let codes_varbin_dtype = codes_varbin.dtype().clone();
+
+        let codes_offsets_delta = DeltaCompressor.compress(
+            &codes_varbin.offsets(),
+            like.as_ref().and_then(|l| l.child(1).cloned()),
+            ctx.auxiliary("offsets"),
+        )?;
```

Comment:
you could completely delegate to the compressor and it would pick the best variant for it